src/magicbox/ResultPreviewsManager.ts

```diff
@@ -0,0 +1,214 @@
+import { $$, Dom } from '../utils/Dom';
+import { l } from '../strings/Strings';
+import { defaults, findIndex } from 'lodash';
+import { Component } from '../ui/Base/Component';
+import { Direction } from './SuggestionsManager';
+
+export interface IPopulateSearchResultPreviewsEventArgs {
+  suggestionText: string;
+  previewsQuery: Promise<ISearchResultPreview[]>;
+}
+
+export enum ResultPreviewsManagerEvents {
+  PopulateSearchResultPreviews = 'populateSearchResultPreviews'
+}
+
+export interface ISearchResultPreview {
+  element: HTMLElement;
+  onSelect: () => void;
+}
+
+export interface IResultPreviewsManagerOptions {
+  previewClass: string;
+  selectedClass: string;
+  previewHeaderText: string;
+}
+
+export class ResultPreviewsManager {
+  private options: IResultPreviewsManagerOptions;
+  private suggestionsPreviewContainer: Dom;
+  private resultPreviewsHeader: Dom;
+  private resultPreviewsContainer: Dom;
+  private lastPreviewsQuery: Promise<ISearchResultPreview[]>;
+  private lastSelectedSuggestion: HTMLElement;
+  private root: HTMLElement;
+
+  public get previewsOwner() {
+    return this.lastSelectedSuggestion;
+  }
+
+  public get hasPreviews() {
+    return !!this.suggestionsPreviewContainer;
+  }
+
+  public get focusedPreviewElement() {
+    if (!this.hasPreviews) {
+      return null;
+    }
+    const focusedElement = this.suggestionsPreviewContainer.findClass(this.options.selectedClass)[0];
+    if (!focusedElement || !focusedElement.classList.contains(this.options.previewClass)) {
+      return null;
+    }
+    return focusedElement;
+  }
+
+  public get previewElements() {
+    if (!this.hasPreviews) {
+      return [];
+    }
+    return this.suggestionsPreviewContainer.findClass(this.options.previewClass);
+  }
+
+  private get suggestionsListbox() {
+    return $$($$(this.element).findClass('coveo-magicbox-suggestions')[0]);
+  }
+
+  private get numberOfResultsPerRow() {
+    const previewSelectables = this.suggestionsPreviewContainer.findClass(this.options.previewClass);
+    if (previewSelectables.length === 0) {
+      return 0;
+    }
+    const firstVerticalOffset = previewSelectables[0].offsetTop;
+    const firstIndexOnNextRow = findIndex(previewSelectables, previewSelectable => previewSelectable.offsetTop !== firstVerticalOffset);
+    return firstIndexOnNextRow !== -1 ? firstIndexOnNextRow : previewSelectables.length;
+  }
```

Comment:
Adaptation of `SuggestionsManager`'s `numberOfResultsPerRow`.
Original version:
```ts
private get numberOfResultsPerRow() {
    const previewSelectables = $$(this.element).findAll('.coveo-preview-selectable');
    if (previewSelectables.length === 0) {
      return 0;
    }
    const firstVerticalOffset = previewSelectables[0].offsetTop;
    const firstIndexOnNextRow = findIndex(previewSelectables, previewSelectable => previewSelectable.offsetTop !== firstVerticalOffset);
    return firstIndexOnNextRow !== -1 ? firstIndexOnNextRow : previewSelectables.length;
}
```